apps/src/weblab/WebLab.js

```diff
@@ -198,6 +218,75 @@ WebLab.prototype.init = function (config) {
   ), document.getElementById(config.containerId));
 };
 
+const PROJECT_URL_PATTERN = /^(.*\/projects\/\w+\/[\w\d-]+)\/.*/;
+/**
+ * @returns the absolute url to the root of this project without a trailing slash.
+ *     For example: http://studio.code.org/projects/applab/GobB13Dy-g0oK
+ */
+function getProjectUrl() {
+  const match = location.href.match(PROJECT_URL_PATTERN);
+  if (match) {
+    return match[1];
+  }
+  return location.href; // i give up. Let's try this?
+}
+
+WebLab.prototype.renderFooterInSharedMode = function (container, copyrightStrings) {
```

Comment:
This function is a close copy of the applab footer. To match the desired UI spec, this may get replaced with something not based on `dashboard.SmallFooter`.